providers/openmetrics/client_interceptor.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	openmetrics "github.com/prometheus/client_golang/prometheus"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
+)
+
+var (
+	// DefaultClientMetrics is the default instance of ClientMetrics. It is
+	// intended to be used in conjunction the default Prometheus metrics
+	// registry.
+	DefaultClientMetrics = NewClientMetrics()
```

Comment:
It would be really nice to avoid adding any global/default things to the v2 of the library! Why not let users decide how to use it? If they use the global registry it can be ~1 line of code to register all metrics with it but if they don't then there is simply no way to opt out. Also, having global as default encourages everybody to use the global metrics but if you have a use case where globals don't work for you, you are having a really hard time because everybody is using them.

For example, currently there is `NewClientMetrics()` so supposedly you can create a new instance and don't use globals, but... there is not way to register metrics inside of the `ClientMetrics` with a custom registry.

Another example - what if you have several clients and some of them should have certain histograms and some of them - should not, or should have a different set of histograms? With globals used everywhere it's impossible to have distinct configuration. I think all `Enable...()` functions should be turned into options for `NewClientMetrics()` to make it all configurable per-instance.

Note that globals can always be added later but cannot be taken out once added (API breakage) so v2 is a perfect time to remove them from the API. Instead `ClientMetrics.Register(prometheus.Registerer)` can be added to make it trivial, whether you use globals or not, to register their metrics with the desired (global or not) registry. Like so:

```go
package my_global_metrics

import (
    "github.com/prometheus/client_golang/prometheus"
    "github.com/grpc-ecosystem/go-grpc-middleware/providers/openmetrics"
)

var (
    DefaultClientMetrics = openmetrics.NewClientMetrics()
)
func init() {
    DefaultClientMetrics.Register(prometheus.DefaultRegisterer).
}
```

Reply:
Yes. That's definitely a plan to get rid of, but we think to do it in another PR. 

However maybe it's easy to do it now?

Reply:
@yashrsharma44 fixing now or in next PR? (: 


I will be available whole day tomorrow so we can quickly iterate.

Reply:
Hey, I will push the changes in this PR itself, by today. Sounds good?

Reply:
Let us know!